plugins/com.aptana.editor.common/src/com/aptana/editor/common/CommonEditorPlugin.java

```diff
@@ -516,4 +536,39 @@ public synchronized IQuickFixProcessorsRegistry getQuickFixProcessorRegistry()
 		}
 		return quickFixRegistry;
 	}
+	
+	/**
+	 * Registering double click listener to the project explorer viewer to handle "Open no extension files with JavaScript Source Editor" preference option.
+	 * Which is added under the "Editors" section. Based on the preference option while opening a file will overwrite the default editor id of a file.
+	 */
+	private void handleOpenWithJSEditorPref()
+	{
+		IViewPart explorerPart = UIUtils.findView(IPageLayout.ID_PROJECT_EXPLORER);
+		if (explorerPart != null)
+		{
+			CommonNavigator projectExplorer = (CommonNavigator) explorerPart;
+			CommonViewer viewer = projectExplorer.getCommonViewer();
+			viewer.addDoubleClickListener(new IDoubleClickListener()
+			{
+				public void doubleClick(DoubleClickEvent event)
+				{
+					IResource selectedResource = UIUtils.getSelectedResource();
+					if (selectedResource != null && selectedResource instanceof IFile)
+					{
+						IFile selectedFile = (IFile) selectedResource;
+						if (!selectedFile.getName().contains(".")) { ////$NON-NLS-N$
+
+							boolean isOpenWithJSEditor = Platform.getPreferencesService().getBoolean(
+									CommonEditorPlugin.PLUGIN_ID,
+									com.aptana.editor.common.preferences.IPreferenceConstants.OPEN_WITH_JS_EDITOR,
+									false, null);
+							IDE.setDefaultEditor(selectedFile, isOpenWithJSEditor ? ICommonConstants.JS_EDITOR_ID
+									: EditorsUI.DEFAULT_TEXT_EDITOR_ID);
+
```

Comment:
I'm not sure whether this approach works 100%. We are trying to hack around the click event listener to set the default editor to JS instead of the base editor. What would happen if there is a timing window and Eclipse has already open the file with the base editor before we set the default editor ?